cli/main.go

```diff
@@ -225,6 +250,118 @@ func Main() {
 	log.Fatal(service.StartHTTPServer())
 }
 
+func checkProposerConfigOptions() error {
+	const (
+		flagRelay              = "relay"
+		flagRelayURLs          = "relays"
+		flagProposerConfigURL  = "proposer-config-url"
+		flagProposerConfigFile = "proposer-config-file"
+	)
+
+	allowedOptions := map[string]string{
+		flagRelay:              relays.String(),
+		flagRelayURLs:          *relayURLs,
+		flagProposerConfigURL:  *proposerConfigURL,
+		flagProposerConfigFile: *proposerConfigFile,
+	}
+
+	providedOptions := make(map[string]string, len(allowedOptions))
+
+	addOptionIfNotEmpty := func(opt, val string) {
+		if _, ok := allowedOptions[opt]; ok && val != "" {
+			providedOptions[opt] = val
+		}
+	}
+
+	for opt, value := range allowedOptions {
+		addOptionIfNotEmpty(opt, value)
+	}
+
+	if len(providedOptions) < 1 {
+		return fmt.Errorf("%w: please specify %s",
+			ErrRequiredOptionsAreNotProvided, mapKeysToString(allowedOptions))
+	}
+
+	if len(providedOptions) > 1 {
+		return fmt.Errorf("%w: please specify %s",
+			ErrMutuallyExclusiveOptions, mapKeysToString(providedOptions))
+	}
+
+	return nil
+}
+
+func mapKeysToString(m map[string]string) string {
+	res := make([]string, 0, len(m))
+	for opt := range m {
+		res = append(res, opt)
+	}
+
+	return "-" + strings.Join(res, " or -")
```

Comment:
For a consistent error message:
```suggestion
	sort.Strings(res)
	return "-" + strings.Join(res, " or -")
```

Without this they will show up in a random order. Also, I would prefer they were in this order:

1. `-relay`
2. `-relays`
3. `-proposer-config-file`
4. `-proposer-config-url`

But that would be more complicated & I don't believe it's worth it.

Reply:
Fixed.